implemented ui for community-link card

izuchukwu-eric opened the pull request:
fixes #14123 

### Summary

[comment]: # (Summarise the problem and how the pull)
created a reusable component for community-link card

#### Platforms
<!-- (Optional. Specify which platforms should be tested) -->

- Android
- iOS
- macOS
- Linux
- Windows

### Steps to test
<!-- (Specify exact steps to test if there are such) -->

- Open Status
- Open community
- Should see a community card for decentraland

status: ready <!-- Can be ready or wip -->

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
ui-component community-link-card

Commit:
fix conflict

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

Commit:
changed text/text to rn/text

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 6762f54a | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-14368/1/) | 2022-11-15 06:57:51 | ~27 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-14368/1/consoleText) |

jo-mut commented on src/quo2/components/community/community_card.cljs:
```diff
@@ -0,0 +1,197 @@
+     ]
+    ]
+
+   [rn/view
```
@izuchukwu-eric align your clojure maps properly

 you can have a look at this doc to see more on how to format your code
https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md

jo-mut commented on translations/en.json:
```diff
@@ -450,6 +450,11 @@
+    "decent-desc": "The first decentralized metaverse that is built, governed, and owned by its users",
+    "decent-title": "Decentraland",
+    "member-count": "629.2k",
+    "likes-count": "112.1k",
```
active-members instead of likes-count
mutual-contacts instead of dislike-count

jo-mut commented on src/quo2/components/community/community_card.cljs:
```diff
@@ -0,0 +1,197 @@
+              :align-self :center
+              :align-items :center
+              :justify-content :center
+              :background-color "#EC4D57"
```
use the `quo2.foundations.colors` component for colors

jo-mut commented on src/quo2/components/community/community_card.cljs:
```diff
@@ -0,0 +1,197 @@
+                  :padding-left 10
+                  :color colors/white
+                  }}
+      "View Community"
```
use `i18n/label`

jo-mut commented on src/status_im/ui2/screens/communities/communities_home.cljs:
```diff
@@ -101,13 +103,26 @@
      [title-column]
      [discover-card]
      [home-community-segments]
+     [community-card]
```
Add your component for preview to `status-im.ui2.screens.quo2-preview.` It should not be here.
See examples of how other components in for instance: 
`status-im.ui2.screens.quo2-preview.community.community-card-view` are added to quo2_preview.

jo-mut commented on src/status_im/ui2/screens/communities/communities_home.cljs:
```diff
@@ -101,13 +103,26 @@
                                 :accessibility-label :communities-home-discover-card}])
 
+
+(defn community-card []
```
`active-members` instead of `likes-count`
`mutual-contacts` instead of ` dislike-count`

jo-mut commented on src/quo2/components/community/community_card.cljs:
```diff
@@ -0,0 +1,197 @@
+    ]
+   ])
+
+(defn community-stats [member_count likes_count dislike_count]
```
better to use a keyword argument here
`community-stats [{:keys [member_count likes_count dislike_count]}]`

jo-mut reviewed (changes requested):
have a look at these guideline https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md and format your code appropriately

izuchukwu-eric commented:
Alright I'll make the due changes...thanks

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

Commit:
added necessary changes

jo-mut commented on src/status_im/ui2/screens/quo2_preview/community/community_link_card.cljs:
```diff
@@ -0,0 +1,24 @@
+            [quo2.components.community.community-link-card :as community-link-card]))
+
+(defn cool-preview []
+    [rn/view
```
I dont think the parent `rn/view` is necessary. is it?

jo-mut commented on src/quo2/foundations/colors.cljs:
```diff
@@ -136,6 +136,10 @@
+
+;;Community link card
+(def community-link-card-button "#EC4D57")
+
```
are the comments needed here?

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+              :height 16}}]
+
+    [rn/view
+     {:style {:margin-left 5}}
+     [text/text {:font-size    15
+                 :font-weight  600
+                 :font-style   :normal
+                 :weight       :semi-bold}
```
Why is there two font-weight attributes ?

izuchukwu-eric commented:
okay I'll remove the unnecessary code and comment

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+
+   [rn/view
+    {:style {:padding-left 15}}
+    [text/text {:font-size       13
+                :font-style      :normal
+                :font-weight     400
+                :letter-spacing  -0.003
+                :weight          :regular}
```
Two weights again
And for font-size 13 we use this attribute
`:size` `:paragraph-2`

Parveshdhull replied:
we don't need to pass font-size, style, spacing etc. Please checkout available values for size in [text.cljs](https://github.com/status-im/status-mobile/blob/433f9185c8e89dffac95a0c3a188bf1ceef2b292/src/quo2/components/markdown/text.cljs#L17)

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+                :weight          :regular}
+     description]]])
+
+(defn community-stats [{:keys [member_count active_members mutual_contacts]}]
```
Clojure naming convention is kebab case
`member-count` `active-members` etc

J-Son89 replied:
same here, update variable names to use `-` instead of `_` for consistency

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+    [text/text {:font-size    13
+                :font-weight  400
+                :font-style   :normal}
+     mutual_contacts]]])
```
Kebab case

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+    [text/text {:font-size    13
+                :font-weight  400
+                :font-style   :normal}
+     active_members]]
```
kebab

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+       :color  colors/neutral-50
+       :left   10}]]
+
+    [text/text {:font-size    13
+                :font-weight  400
+                :font-style   :normal}
```
This can be `{:weight` `:regular` `:size` `:paragraph-2}`

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+     mutual_contacts]]])
+
+
+(defn community-link-card [{:keys [title description image member_count active_members mutual_contacts]}]
```
Kebab case

J-Son89 replied:
we don't use `_` for variable names. Can you change these to `member-count`, `active-members` etc

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+                  {:height           176
+                   :width            350
+                   :align-self       :center
+                   :padding-bottom   12
+                   :padding-top      12
```
Can be `padding-vertical`

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+             :justify-content    :center
+             :background-color   colors/community-link-card-button
+             :border-radius      10
+             :padding-top        5
+             :padding-right      12
+             :padding-bottom     5
+             :padding-left       12
```
Can be padding vertical and padding horizontal

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+       {:size 20
+        :color colors/white}]]
+
+     [rn/text {:style {:font-size      15
+                       :font-weight    500
+                       :font-style     :normal
+                       :padding-left   10
+                       :color          colors/white}}
+      (i18n/label :t/community-link-card-button-text)]]])
```
Kindly use text component from `quo2.text`

ibrkhalil commented on src/status_im/ui2/screens/communities/communities_home.cljs:
```diff
@@ -101,6 +101,7 @@
                                 :description         (i18n/label :t/whats-trending)
                                 :accessibility-label :communities-home-discover-card}])
 
+
```
Unneeded change

Parveshdhull commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+    [rn/image
+     {:source image
+      :style {:width  16
+              :height 16}}]
```
1. first `[rn/view` can be replaced with `[:<>`
2. Please remove unnecessary empty lines

ibrkhalil commented on src/status_im/ui2/screens/quo2_preview/community/community_link_card.cljs:
```diff
@@ -0,0 +1,24 @@
+       [rn/flat-list {:flex                      1
+                      :keyboardShouldPersistTaps :always
+                      :header                    [cool-preview]
+                      :key-fn                    str}]])
```
Kindly add a new line at EOF

ibrkhalil commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+            :width             271
+            :padding-left      15}}
+
+   [rn/view
+    {:style {:flex-direction  :row
+             :align-items     :center}}
+
+    [rn/view
+     [icons/icon  :i/add-user
+      {:size    20
+       :color   colors/neutral-50}]]
+
+    [text/text {:font-size    13
+                :font-weight  400
+                :font-style   :normal}
+     member_count]]
+
+   [rn/view
+    {:style {:flex-direction      :row
+             :align-items         :center
+             :padding-horizontal  10}}
+
+    [rn/view
+     [icons/icon  :i/add-user
+      {:size   20
+       :color  colors/neutral-50
+       :left   10}]]
```
Kindly make sure that `rn/view` is needed .. As too many redundant `rn/view` usages are bad for performance

Commit:
removed unnecessary code & comment

Commit:
removed unnecessary code & comment

Commit:
removed unnecessary code & comment

Commit:
removed unnecessary code & comment

Commit:
removed unnecessary code & comment

Commit:
removed unnecessary code & comment

J-Son89 commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+    [rn/view
+     {:style {:margin-left 5}}
+     [text/text {:font-size    15
+                 :font-weight  600
```
hmm, perhaps you should be using
```
[text/text {:size  :paragraph-1 
:weight :semi-bold
}]
```
see the file: "src/quo2/foundations/typography.cljs"

for a the different sizes.

Unless there's a specific reason you need these exact props

J-Son89 commented on src/quo2/components/community/community_link_card.cljs:
```diff
@@ -0,0 +1,136 @@
+
+   [rn/view
+    {:style {:padding-left 15}}
+    [text/text {:font-size       13
```
same here, could this be:

```
[text/text {:size  :paragraph-2
}]
```
`:weight :regular` is the default as far as I know. Best to double check that

J-Son89 commented:
If possible it would be great if you can add a screen shot or two of the implementation.

and when attaching them to github comments you can use 
<img src="..." height="400" /> or any other combination of width/height 
where src is the url github displays when you upload. (This helps avoid the image being huge and taking over the p.r feed)

izuchukwu-eric commented:
thanks for the review everyone, will send an updated commit

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

Commit:
made requested changes

izuchukwu-eric commented:
Alright will do that
I added `:i/group` and `:i/lightning`  icons, but for some reason they aren't showing...any tip?
I tried taking a screenshot but due to the privacy policy it doesn't work

Commit:
ran lint-fix

Commit:
ran lint-fix

Commit:
ran lint-fix

Commit:
ran lint-fix

Commit:
ran lint-fix

Commit:
ran lint-fix

jo-mut commented:
You can restart `make run-clojure` to see new added icons

izuchukwu-eric commented:
Alright...thanks

izuchukwu-eric commented:
> You can restart `make run-clojure` to see new added icons

restarting `make run-clojure` doesn't seem to fix it

Commit:
fixed icon issue

Commit:
fixed icon issue

Commit:
fixed icon issue

Commit:
fixed icon issue

Commit:
fixed icon issue

Commit:
fixed icon issue

izuchukwu-eric commented:
![Screenshot_20221117-000432_Status Debug](https://user-images.githubusercontent.com/57435925/202315517-8db13926-6d99-46af-938e-c6487ccc6778.jpg)
This is how it looks

izuchukwu-eric commented:
![Screenshot_20221117-000432_Status Debug](https://user-images.githubusercontent.com/57435925/202315741-3e4d21e4-3961-46f1-a954-fd3db989cd94.jpg)
This is how it looks

jo-mut commented:
looks good to me but your tests need to pass. ran `make lint-fix` & `make-lint`

izuchukwu-eric committed:
Merge branch 'develop' into feature/community-link-card

izuchukwu-eric commented:
> looks good to me but your tests need to pass. ran `make lint-fix` & `make-lint`

running `make lint` says the files are indented correctly

izuchukwu-eric commented:
> > > looks good to me but your tests need to pass. ran `make lint-fix` & `make-lint`
> > 
> > 
> > running `make lint` says the files are indented correctly
> 
> make sure you commit and push your changes

I already made a commit that fixes the lint issues. could there be any other reasons why the commits aren't passing the jenkins test?

churik commented:
@izuchukwu-eric yes, there is a conflict on rebase stage:
```
18:25:40  Rebasing (1/7)
CONFLICT (modify/delete): src/status_im/ui2/screens/communities/communities_home.cljs deleted in HEAD and modified in b082f7782 (ui-component community-link-card).  Version b082f7782 (ui-component community-link-card) of src/status_im/ui2/screens/communities/communities_home.cljs left in tree.
18:25:40  Auto-merging translations/en.json
18:25:40  error: could not apply b082f7782... ui-component community-link-card
18:25:40  hint: Resolve all conflicts manually, mark them as resolved with
18:25:40  hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
18:25:40  hint: You can instead skip this commit: run "git rebase --skip".
18:25:40  hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
18:25:40  Could not apply b082f7782... ui-component community-link-card
```

Commit:
rebasing
Merge branch 'develop' of https://github.com/izuchukwu-eric/status-mobile into feature/community-link-card

Commit:
fixed linting issue

izuchukwu-eric commented:
> @izuchukwu-eric yes, there is a conflict on rebase stage:
> 
> ```
> 18:25:40  Rebasing (1/7)
> CONFLICT (modify/delete): src/status_im/ui2/screens/communities/communities_home.cljs deleted in HEAD and modified in b082f7782 (ui-component community-link-card).  Version b082f7782 (ui-component community-link-card) of src/status_im/ui2/screens/communities/communities_home.cljs left in tree.
> 18:25:40  Auto-merging translations/en.json
> 18:25:40  error: could not apply b082f7782... ui-component community-link-card
> 18:25:40  hint: Resolve all conflicts manually, mark them as resolved with
> 18:25:40  hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
> 18:25:40  hint: You can instead skip this commit: run "git rebase --skip".
> 18:25:40  hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
> 18:25:40  Could not apply b082f7782... ui-component community-link-card
> ```

Thanks for pointing this out, I think its resolved now

izuchukwu-eric committed:
Merge branch 'develop' into feature/community-link-card

izuchukwu-eric committed:
Merge branch 'status-im:develop' into develop

churik commented:
@izuchukwu-eric 
```
[2022-11-18T10:56:55.155Z] + git rebase develop

[2022-11-18T10:56:55.155Z] Rebasing (1/14)
CONFLICT (modify/delete): src/status_im/ui2/screens/communities/communities_home.cljs deleted in HEAD and modified in b082f7782 (ui-component community-link-card).  Version b082f7782 (ui-component community-link-card) of src/status_im/ui2/screens/communities/communities_home.cljs left in tree.

[2022-11-18T10:56:55.155Z] Auto-merging translations/en.json

[2022-11-18T10:56:55.155Z] error: could not apply b082f7782... ui-component community-link-card

[2022-11-18T10:56:55.155Z] hint: Resolve all conflicts manually, mark them as resolved with

[2022-11-18T10:56:55.155Z] hint: "git add/rm <conflicted_files>", then run "git rebase --continue".

[2022-11-18T10:56:55.155Z] hint: You can instead skip this commit: run "git rebase --skip".

[2022-11-18T10:56:55.155Z] hint: To abort and get back to the state before "git rebase", run "git rebase --abort".

[2022-11-18T10:56:55.155Z] Could not apply b082f7782... ui-component community-link-card

script returned exit code 1
```

Commit:
Merge branch 'feature/community-link-card' of https://github.com/izuchukwu-eric/status-mobile into feature/community-link-card

Commit:
Merge branch 'develop' of https://github.com/izuchukwu-eric/status-mobile into develop

Commit:
Merge branch 'feature/community-link-card' of https://github.com/izuchukwu-eric/status-mobile into feature/community-link-card

Commit:
Merge branch 'feature/community-link-card' of https://github.com/izuchukwu-eric/status-mobile into feature/community-link-card

Commit:
Merge branch 'feature/community-link-card' of https://github.com/izuchukwu-eric/status-mobile into feature/community-link-card

Commit:
Merge branch 'feature/community-link-card' of https://github.com/izuchukwu-eric/status-mobile into feature/community-link-card

Commit:
Merge branch 'feature/community-link-card' of https://github.com/izuchukwu-eric/status-mobile into feature/community-link-card

izuchukwu-eric commented:
> @izuchukwu-eric
> 
> ```
> [2022-11-18T10:56:55.155Z] + git rebase develop
> 
> [2022-11-18T10:56:55.155Z] Rebasing (1/14)
> CONFLICT (modify/delete): src/status_im/ui2/screens/communities/communities_home.cljs deleted in HEAD and modified in b082f7782 (ui-component community-link-card).  Version b082f7782 (ui-component community-link-card) of src/status_im/ui2/screens/communities/communities_home.cljs left in tree.
> 
> [2022-11-18T10:56:55.155Z] Auto-merging translations/en.json
> 
> [2022-11-18T10:56:55.155Z] error: could not apply b082f7782... ui-component community-link-card
> 
> [2022-11-18T10:56:55.155Z] hint: Resolve all conflicts manually, mark them as resolved with
> 
> [2022-11-18T10:56:55.155Z] hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
> 
> [2022-11-18T10:56:55.155Z] hint: You can instead skip this commit: run "git rebase --skip".
> 
> [2022-11-18T10:56:55.155Z] hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
> 
> [2022-11-18T10:56:55.155Z] Could not apply b082f7782... ui-component community-link-card
> 
> script returned exit code 1
> ```

I'm getting a rebase successfully message after fixing the conflicts, not why it keeps showing these conflicts even after rebasing. are there any other solutions?
